### PR TITLE
Issue #625: Tugboat urls support query parameters.

### PR DESCRIPTION
**Pull request by mrdavidburns**

### CODE REVIEW

File: src/ScaffoldInstallerPlugin.php [hunk 1]
```diff
@@ -407,19 +407,30 @@ private function installTugboat(string $scaffoldPath): void {
...
-                ARRAY_FILTER_USE_KEY);
+            $tugboatConfigOverride['php'] = array_filter($tugboatConfigOverride['php'], function($key) {
+                return in_array($key, ['aliases', 'urls', 'visualdiff', 'screenshot']);
+            }, ARRAY_FILTER_USE_KEY);
+
+            // Prepare YAML data
+            $yamlData = [];
+            foreach ($tugboatConfigOverride['php'] as $key => $value) {
+                $yamlData[$key] = $value;
+            }
```
Comment (deviantintegral) [inline src/ScaffoldInstallerPlugin.php:418]: This seems wrong - as in, I suppose it works, but don't you just want something like:

```php
$yamlData = $tugboatConfigOverride['php'];
```

File: src/ScaffoldInstallerPlugin.php [hunk 2]
```diff
@@ -407,19 +407,30 @@ private function installTugboat(string $scaffoldPath): void {
...
         if (!empty($tugboatConfigOverride['php']) && is_array($tugboatConfigOverride['php'])) {
-            $tugboatConfigOverride['php'] = array_filter($tugboatConfigOverride['php'],
-                function($key) {
-                    return in_array($key,
-                        ['aliases', 'urls', 'visualdiff', 'screenshot']);
-                },
-                ARRAY_FILTER_USE_KEY);
+            $tugboatConfigOverride['php'] = array_filter($tugboatConfigOverride['php'], function($key) {
+                return in_array($key, ['aliases', 'urls', 'visualdiff', 'screenshot']);
+            }, ARRAY_FILTER_USE_KEY);
```
Comment (deviantintegral) [inline src/ScaffoldInstallerPlugin.php:412]: Just confirming, this appears to only be formatting changes?

**Review by deviantintegral — COMMENTED** [review]
I'm a little confused by this PR.

At the least, I think doing a broad htmldecode on the YAML itself could lead to other bugs. For example, what happens if you actually want an HTML entity in some other text? What if you are inlining HTML?

Reading the code, it's as if `Yaml::dump()` is encoding HTML entities. But, I did a quick test at https://github.com/deviantintegral/yaml-test and it doesn't:

```
/opt/homebrew/bin/php /Users/andrew/workspace/yaml-test/test.php
urls:
  - '/path?query=value'

Process finished with exit code 0
```

Do you think you can narrow this down a bit into something more targeted that tells us exactly what function is doing the HTML encoding? I like @beto-aveiga 's suggestion of xdebug!

**Comment by mrdavidburns** [comment]
Closing in favor of #681

**Pull request closed without merging**